rapid7/metasploit-framework
# Convert MSSQL mixin to class

This continues the work started in https://github.com/rapid7/metasploit-framework/pull/18615 

Before adding an MSSQL session type, we need a standalone client class, in addition to a mixin. This PR consolidates the two instances of MSSQL client classes, consolidates them, and converts the consolidated mixin into a class that all existing MSSQL modules have been tweaked to work with. 

To test: 
- Boot up an MSSQL instance, and take note of the machine's ip
- load up any modules changed in this pr. Set `rhost` to the target's ip, `rport` to the port MSSQL is running on (usually 1433), `username`, `password`, and any other required options.
- run the module
- compare behavior to master. Nothing should have changed
